msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/PublicClientApplication.java

```diff
@@ -216,4 +216,61 @@ protected Builder self() {
             return this;
         }
     }
+
+    /**
+     * Used to determine whether to call into an IBroker instance instead of standard MSAL Java's normal interactive flow,
+     * and may throw exceptions or log messages if broker-only parameters are used when a broker is not enabled/available
+     */
+    private boolean validateBrokerUsage(InteractiveRequestParameters parameters) {
+
+        //Check if broker-only parameters are being used when a broker is not enabled. If they are, either throw an
+        // exception saying a broker is required, or provide a clear log message saying the parameter will be ignored
+        if (!brokerEnabled) {
+            if (parameters.proofOfPossession() != null) {
+                throw new MsalClientException(
+                        "InteractiveRequestParameters.proofOfPossession should not be used when broker is not available, see https://aka.ms/msal4j-pop for more information",
+                        AuthenticationErrorCode.MSALJAVA_BROKERS_ERROR );
+            }
+        }
+
+        return brokerEnabled;
+    }
+
+    /**
+     * Used to determine whether to call into an IBroker instance instead of standard MSAL Java's normal username/password flow,
+     * and may throw exceptions or log messages if broker-only parameters are used when a broker is not enabled/available
+     */
+    private boolean validateBrokerUsage(UserNamePasswordParameters parameters) {
```

Comment:
On top of this, the spec states to have an API like "IsPoPSupportedByClient" so that clients can have logic like 

- does RP support POP and does client support POP?

I suppose that `IsBrokerSupported` is public API and it can be used for the same?

Reply:
Yes, there is an 'isBrokerAvailable' API that essentially just answers the question "could the MSALRuntime startup API be called successfully". 

Unless there is some scenario where POP isn't supported but MSALRuntime works, then for MSAL Java at least an `isBrokerSupported` API and an `isPoPSupported` API would always return the same thing.